### PR TITLE
Implement new get building period() method #15

### PR DESCRIPTION
**Pull request by longshuicy**
TO test:

run test for `building damage`, `non-structural building damage` and `pipeline damage`

### CODE REVIEW

**Comment by navarroc** [comment]
Can you point this at the release branch?

**Comment by navarroc** [comment]
Tested this change and it looks good. I think we can remove the hack on line 458 though, right? We only hit that line because the fragilitycurve was always returning 0.0 for get_building_period.

**Review by navarroc — APPROVED** [review]

File: pyincore/analyses/buildingdamage/buildingdamage.py [hunk 1]
```diff
@@ -171,7 +171,6 @@ def building_damage_analysis_bulk_input(self, buildings, retrofit_strategy, haza
             num_stories = b['properties']['no_stories']
```
Comment (navarroc) [inline pyincore/analyses/buildingdamage/buildingdamage.py:171]: I think this is unused now

File: pyincore/analyses/buildingdamage/buildingdamage.py [hunk 2]
```diff
@@ -171,7 +171,6 @@ def building_damage_analysis_bulk_input(self, buildings, retrofit_strategy, haza
...
             selected_fragility_set = fragility_sets[b_id]
```
Comment (navarroc) [inline pyincore/analyses/buildingdamage/buildingdamage.py:173]: Extra blank line

**Comment by navarroc** [comment]
analysisutil has an unused import.

**Comment by longshuicy** [comment]
> analysisutil has an unused import.

removed

File: pyincore/models/fragilitycurve.py [hunk 3]
```diff
@@ -101,31 +101,21 @@ def calculate_limit_state_probability(self, hazard_values: dict, fragility_curve
...
         return probability
 
-    def get_building_period(self, num_stories):
-        """Get building period from the fragility curve.
-        Args:
-            num_stories (int): Number of building stories.
-        Returns:
-            float: Building period.
-        """
+    def get_building_period(self, fragility_curve_parameters, **kwargs):
```
Comment (ncsa-mo) [inline pyincore/models/fragilitycurve.py:104]: Can you add these docstings here:

def get_building_period(self, fragility_curve_parameters, **kwargs):
        """ Get building period from the fragility curve.
        
        Args:
            fragility_curve_parameters (dict): Fragility curve parameters.
            **kwargs: Keyword arguments.
        
        Returns:
            float: Building period.
            
        """

**Review by ncsa-mo — APPROVED** [review]

**Review by diegoac2 — APPROVED** [review]

**Pull request merged**